src/Processors/Formats/Impl/TSKVRowInputFormat.cpp

```diff
@@ -242,15 +241,16 @@ std::unordered_map<String, DataTypePtr> TSKVSchemaReader::readRowAndGetNamesAndD
 
     std::unordered_map<String, DataTypePtr> names_and_types;
     StringRef name_ref;
-    String name_tmp;
+    String name;
     String value;
     do
     {
-        bool has_value = readName(in, name_ref, name_tmp);
+        bool has_value = readName(in, name_ref, name);
+        name = String(name_ref);
```

Comment:
```suggestion
        bool has_value = readName(in, name_ref, name_buf);
        String name = String(name_ref);
```

You clear `name_buf` anyway in `readName`.